RickLeinecker/Fortuna
# Api battles

**Description**
Created a setWager, prepareBattle, and reportResults API. The setWager api simply updates the wager field of the user model. The prepareBattle api creates a BattleRecord with the user's and tanks and returns the BattleRecord._id. The reportResults api records the winner and updates the participating users' money, wager, and stats and returns the completed BattleRecord.

Resolves #228 
Resolves #202
Resolves #229 

**Type of change**
Check options that are relevant.


**Steps to Verify Functionality**
setWager API - requires auth token in header
1) Call the setWager api and give it the wager amount
2) Check if the user doc was updated accordingly

prepareBattle API - requires auth token in header
1) Create two users
2) Give the challengee a favorite tank
3) Give the userId of the challengee and the tankId of the challenger to the api in the body
4) Check to see if a battle record was created in the db with the filled out fields

reportResults API - requires auth token in header
1) Give the api the id of the battle record you received from the prepareBattle API and the winner represented by 0 for a tie, 1 for personBeingChallenged victory, 2 for challenger victory all through the body
2) Check if the associated battle record doc and user docs were updated accordingly

**Final Checklist:**
Go down the list and check them off.
